### PR TITLE
Import basic orderly.yml reading

### PR DESCRIPTION
**Pull request by richfitz**
Towards a proof of concept orderly-like system built with outpack. Lots to add here, but aiming for a really basic proof-of-concept that can handle running a really simple report.

First, the reading is directly copied over from orderly itself, stripped down as much as I can to support the minimum features.

See read.R for bits that are not supported yet - I've tried to get ahead of features that we'll implement a bit, while keeping it hopefully manageableish. This is not new code though, just copied over code from orderly (but with some slightly better testing I think)

### CODE REVIEW

File: R/root.R [hunk 1]
```diff
@@ -0,0 +1,60 @@
+## In orderly, there's quite a bit more to read here, coping with the
+## configuration (orderly_config.yml) which also marks the root.
+##
+## In orderly, the key fields in the configuration are:
```
Comment (r-ash) [inline R/root.R:4]: old comment here? There are no items in the list

File: R/root.R [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  ## "draft" for this to work well.
+  path <- root$path
+  config <- orderly_config(path)
+  ret <- list(outpack = root, config = config, path = path)
+  class(ret) <- "orderly_root"
+  ret
+}
+
+
+orderly_init <- function(path) {
```
Comment (r-ash) [inline R/root.R:38]: You don't want to create the `src` dir as part of this? Or maybe just something to iterate on later, I expect there is other setup to do

**Review by r-ash — APPROVED** [review]
Looks good, couple of minor questions/comments.

File: R/root.R (hunk 1 above)
Comment (richfitz) [reply R/root.R:4]: the next three subheadings are in the list!

File: R/root.R (hunk 2 above)
Comment (richfitz) [reply R/root.R:38]: The latter yeah - I just want the root here so that we can use it for reading atm

**Pull request merged**